backend/debug/dwarf/dwarf_ocaml/dwarf_concrete_instances.ml

```diff
@@ -0,0 +1,52 @@
+[@@@ocaml.warning "+a-4-30-40-41-42"]
+
+open Asm_targets
+open Dwarf_high
+
+type fundecl =
+  { fun_name : string;
+    fun_dbg : Debuginfo.t
+  }
+
+module DAH = Dwarf_attribute_helpers
+
+let end_symbol_name start_symbol = start_symbol ^ "__end"
+
+let for_fundecl ~get_file_id state fundecl =
+  let parent = Dwarf_state.compilation_unit_proto_die state in
+  let fun_name = fundecl.fun_name in
+  let linkage_name =
+    match fundecl.fun_dbg with
+    | [item] -> Debuginfo.Scoped_location.string_of_scopes item.dinfo_scopes
+    (* Not sure what to do in the cases below *)
```

Comment:
This should probably be fixed so that if there isn't a single `item`, we just use `fun_name`.  I'm unsure this will ever happen in practice.